modulegen/internal/dependabot/writer.go

```diff
@@ -0,0 +1,23 @@
+package dependabot
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+func writeConfig(configFile string, config *Config) error {
+	err := os.MkdirAll(filepath.Dir(configFile), 0o755)
+	if err != nil {
+		return fmt.Errorf("create directory: %w", err)
+	}
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("marshal config: %w", err)
+	}
+	header := "# This file is autogenerated by the 'modulegen' tool.\n# Please look at the generator code when updating it.\n"
```

Comment:
suggestion: use the standard header format detailed [here](https://github.com/golang/go/issues/13560#issuecomment-288457920)

Reply:
But the generated file in this case it not a Go file, but the dependabot.yml descriptor. We should use the proper comment character.

Reply:
There might be an added benefit, would need to check, but editors and github understood this header when I last checked, although that might just be for go files as you mentioned.